service/block/erasure.go

```diff
@@ -0,0 +1,33 @@
+package block
+
+import (
+	"math"
+
+	"github.com/celestiaorg/celestia-core/pkg/wrapper"
+	"github.com/celestiaorg/rsmt2d"
+)
+
+// extendBlockData erasure codes the given raw block's data and returns the
+// erasure coded block data upon success.
+func (s *Service) extendBlockData(raw *Raw) (*ExtendedBlockData, error) {
```

Comment:
Is there a particular reason this is a method that has the Service as a receiver? Should this not be just a function free from side-effects on the Service?

Reply:
Yes I think it could be standalone. I will keep it private though because no service or process outside the block service package should perform erasure coding.

Reply:
Alternatively, could make it a method on `Raw`. 

`raw.extendBlockData()`

Reply:
Nvm that would be annoying bc `Raw` would no longer be just an alias to core.Block.

Reply:
Exactly, I think just making this function instead of a method is the best approach.